src/main/java/io/pravega/connectors/hadoop/PravegaInputSplit.java

```diff
@@ -40,66 +38,74 @@ public PravegaInputSplit() {
     /**
      * Creates an InputSplit corresponding to a Pravega segment.
      *
-     * @param segment     The pravega segment
-     * @param startOffset start offset in the segment
-     * @param endOffset   end offset in the segment
+     * @param segmentRange     The pravega SegmentRange
      */
-    public PravegaInputSplit(Segment segment, long startOffset, long endOffset) {
-        this.segment = segment;
-        this.startOffset = startOffset;
-        this.endOffset = endOffset;
+    public PravegaInputSplit(SegmentRange segmentRange) {
+        this.segmentRange = segmentRange;
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
     public void readFields(DataInput in) throws IOException {
-        segment = Segment.fromScopedName(Text.readString(in));
-        startOffset = in.readLong();
-        endOffset = in.readLong();
+        String scopeName = Text.readString(in);
+        String streamName = Text.readString(in);
+        int segmentNumber = in.readInt();
+        long startOffset = in.readLong();
+        long endOffset = in.readLong();
+        this.segmentRange = SegmentRangeImpl.builder()
+            .segment(new Segment(scopeName, streamName, segmentNumber))
+            .startOffset(startOffset)
+            .endOffset(endOffset).build();
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
     public void write(DataOutput out) throws IOException {
-        Text.writeString(out, segment.getScopedName());
-        out.writeLong(startOffset);
-        out.writeLong(endOffset);
+        Text.writeString(out, this.segmentRange.getScope());
+        Text.writeString(out, this.segmentRange.getStreamName());
+        out.writeInt(this.segmentRange.getSegmentNumber());
+        out.writeLong(this.segmentRange.getStartOffset());
+        out.writeLong(this.segmentRange.getEndOffset());
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
     public int compareTo(PravegaInputSplit o) {
-        int res = segment.compareTo(o.getSegment());
+        int res = this.getSegment().compareTo(o.getSegment());
         if (res == 0) {
-            res = Long.compare(startOffset, o.getStartOffset());
+            res = Long.compare(this.getStartOffset(), o.getStartOffset());
         }
         if (res == 0) {
-            res = Long.compare(endOffset, o.getEndOffset());
+            res = Long.compare(this.getEndOffset(), o.getEndOffset());
         }
         return res;
     }
 
     public Segment getSegment() {
```

Comment:
we can narrow down the access modifiers, right?

Reply:
good catch! accepted.